lib/src/pgn.dart

```diff
@@ -0,0 +1,831 @@
+                  if (_stack.length == 1 && token != '*') {
+                    _gameHeaders['Result'] = token;
+                  }
+                } else if (token == '(') {
+                  _stack.add(_ParserFrame(parent: frame.parent, root: false));
+                } else if (token == ')') {
+                  if (_stack.length > 1) _stack.removeLast();
+                } else if (token == '{') {
+                  final openIndex = match.end;
+                  final beginIndex =
+                      line[openIndex] == ' ' ? openIndex + 1 : openIndex;
+                  line = line.substring(beginIndex);
+                  _state = _ParserState.comment;
+                  continue continuedLine;
+                } else {
+                  if (token == 'Z0' || token == '0000' || token == '@@@@') {
+                    token = '--';
+                  } else if (token.startsWith('0')) {
+                    token = token.replaceAll('0', 'O');
+                  }
+                  if (frame.node != null) {
+                    frame.parent = frame.node!;
+                  }
+                  frame.node = PgnChildNode(PgnNodeData(
+                      san: token, startingComments: frame.startingComments));
+                  frame.startingComments = null;
+                  frame.root = false;
+                  frame.parent.children.add(frame.node!);
+                }
+              }
+            }
+            return;
+          }
+
+        case _ParserState.comment:
+          {
+            final closeIndex = line.indexOf('}');
+            if (closeIndex == -1) {
+              _commentBuf.add(line);
+              return;
+            } else {
+              final endIndex = closeIndex > 0 && line[closeIndex - 1] == ' '
+                  ? closeIndex - 1
+                  : closeIndex;
+              _commentBuf.add(line.substring(0, endIndex));
+              _handleComment();
+              line = line.substring(closeIndex);
+              _state = _ParserState.moves;
+              freshLine = false;
+            }
+          }
+      }
+    }
+  }
+
+  void _handleNag(int nag) {
+    final frame = _stack[_stack.length - 1];
+    if (frame.node != null) {
+      frame.node!.data = frame.node!.data.copyWith(nag: nag);
+    }
+  }
+
+  void _handleComment() {
+    final frame = _stack[_stack.length - 1];
+    final comment = _commentBuf.join('\n');
+    _commentBuf = [];
+    if (frame.node != null) {
+      frame.node!.data = frame.node!.data.copyWith(comment: comment);
+    } else if (frame.root) {
+      _gameComments.add(comment);
+    } else {
+      frame.startingComments ??= [];
+      frame.startingComments!.add(comment);
+    }
+  }
+}
+
+/// Default function to parse a multi game PGN
+///
+/// Returns a list of games if multiple games found
+/// Provide a optional function [initHeaders] to create different headers other than the default
+List<PgnGame<PgnNodeData>> parseMultiGamePgn(String pgn,
```

Comment:
Same as `PgnGame.parse`; check the potential error, and throws a PgnError if needed; also add it can potentially throws in the doc commment.